torchgeo/datasets/spacenet1.py

```diff
@@ -0,0 +1,215 @@
+        for imgpath in images:
+            rawpath = imgpath.replace("RGB.tif", "8Band.tif")
+            lbl_path = os.path.join(
+                os.path.dirname(rawpath) + "-labels", "labels.geojson"
+            )
+            files.append({"rgb": imgpath, "8band": rawpath, "label": lbl_path})
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        filename = os.path.join(path)
+        with rio.open(filename) as img:
+            array = img.read().astype(np.float32)
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor, img.transform
+
+    def _load_label(self, path: str, tfm: Affine, shape: Tuple[int, int]) -> Tensor:
+        """Rasterizes the dataset's labels (in geojson format).
+
+        Args:
+            path (str): path to the label
+            tfm (Affine): transform of corresponding image
+            shape (List[int, int]): shape of corresponding image
+
+        Returns:
+            Tensor: label tensor
+        """
+
+        with fiona.open(path) as src:
+            labels = [feature["geometry"] for feature in src]
+
+        if not labels:
+            mask_data = np.zeros(shape=shape)
+        else:
+            mask_data = rasterize(
+                labels,
+                out_shape=shape,
+                fill=0,  # nodata value
+                transform=tfm,
+                all_touched=False,
+                dtype=np.uint8,
+            )
+
+        mask: Tensor = torch.from_numpy(mask_data).long()  # type: ignore[attr-defined]
+
+        return mask
+
+    def __len__(self) -> int:
+        """Return the number of samples in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        rgb, tfm = self._load_image(files["rgb"])
+        raw, _ = self._load_image(files["8band"])
+        out_shape = tuple(rgb.shape[1:])
+        label = self._load_label(files["label"], tfm, out_shape)
+
+        sample = {"rgb": rgb, "8band": raw, "label": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _check_integrity(self) -> bool:
+        """Checks the integrity of the dataset structure.
+        Returns:
+            True if the dataset directories are found, else False
+        """
+        stacpath = os.path.join(self.root, self.foldername, "collection.json")
+        # If dataset folder does not exist. Check for uncorrupted archive
+        if not bool(os.path.exists(stacpath)):
```

Comment:
Does this need to be wrapped as a `bool`? `os.path.exists` should already return a True/False, no?